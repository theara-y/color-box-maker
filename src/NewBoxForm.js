import { useState } from "react";

const NewBoxForm = ({ addBox }) => {
    const INIT_FORM = {
        width: "",
        height: "",
        color: ""
    }
    const [formData, setFormData] = useState(INIT_FORM);

    const handleChange = (e) => {
        setFormData(oldFormData => {
            const { name, value } = e.target;
            return { ...oldFormData, [name]: value }
        });
    };

    const submit = (e) => {
        e.preventDefault();
        if (Object.values(formData).every(value => value != "")) {
            addBox({...formData});
            setFormData(INIT_FORM)
        }
    }

    return (
        <form onSubmit={submit}>
            <label htmlFor="color">Color</label>
            <input id="color" name="color" type="text" value={formData.color} onChange={handleChange} />
            <label htmlFor="width">Width</label>
            <input id="width" name="width" type="number" min="10" max="300" value={formData.width} onChange={handleChange} />
            <label htmlFor="height">Height</label>
            <input id="height" name="height" type="number" min="10" max="300" value={formData.height} onChange={handleChange} />
            <button>Create Box</button>
        </form>
    );
};

export default NewBoxForm;