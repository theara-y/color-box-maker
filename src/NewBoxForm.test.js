import { fireEvent, getByLabelText, render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

test('render NewBoxForm', () => {
    render(<NewBoxForm/>);
});

test('NewBoxForm snapshot', () => {
    const { asFragment } = render(<NewBoxForm/>);
    expect(asFragment()).toMatchSnapshot()
});