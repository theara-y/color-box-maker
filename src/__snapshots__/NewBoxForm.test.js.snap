// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color
    </label>
    <input
      id="color"
      name="color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      max="300"
      min="10"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      max="300"
      min="10"
      name="height"
      type="number"
      value=""
    />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
